src/core.jsx

```diff
@@ -91,7 +92,13 @@ export default (
 
     render() {
       if (!this.isLoaded()) {
-        return <LoadingIndicator {...this.state.props} />
+        if (LoadingIndicator) {
```

Comment:
I tried this implem : 
```es6
      if (!this.isLoaded()) {
        // No loading indicator : go away
        if (!LoadingIndicator) return null

        // Loading indicator is a component
        if (LoadingIndicator.prototype.isReactComponent) {
          return (
            <LoadingIndicator
              className={this.state.props.className}
              style={this.state.props.style}
            />
          )
        }

        // LoadingIndicator is a function
        // eslint-disable-next-line new-cap
        return LoadingIndicator(this.state.props)
      }
```

But it doesn't work with react pure function components : 
Test `should print a different LoadingIndicator component` fail with this : 
```es6
  const LoadingIndicator = () => <div />
```